### ISSUE TITLE
Itsycal 11.14 cannot be opened

### ISSUE THREAD

**Issue by robmonte** [sfsam/Itsycal#125]
Note: I am on MacOS 10.14.4.

After updating to Itsycal 11.14, the application no longer works for me. The only error message I get is:
"The application “Itsycal.app” can’t be opened."

I tried doing chmod +x on the executable inside the .app file and it doesn't fix it, instead the OS says Itsycal is damaged and should be deleted.

I really like this clock/calendar replacement, I am dying without it working.

**Comment by sfsam** [comment]
Can you delete the app entirely and download it again?

**Comment by robmonte** [comment]
Still happens. Also tried to install it on a new iMac that has never had it before and the same exact thing happens.

**Comment by sfsam** [comment]
Does an older version work?

**Comment by sfsam** [comment]
Here are some urls for older versions:
https://s3.amazonaws.com/itsycal/Itsycal-0.11.13.zip
https://s3.amazonaws.com/itsycal/Itsycal-0.11.12.zip

**Comment by robmonte** [comment]
Yes, 11.13 works again just fine. Interestingly, if I update to 11.14 through the built in "Check for updates" while running 11.13, then it installs and runs 11.14 correct. On the iMac also.

So it is just happening when manually downloading/installing it.

**Comment by sfsam** [comment]
So neither of these work?
https://s3.amazonaws.com/itsycal/Itsycal.zip
https://s3.amazonaws.com/itsycal/Itsycal-0.11.14.zip

**Comment by robmonte** [comment]
The top one does not work.

The bottom one does. Is it possibly you accidentally uploaded a bad build without the version number, being the top URL?

**Comment by sfsam** [comment]
They all work for me, so I don't know what the problem is. I'd like to figure this out, but right now I'm stumped.

**Comment by sfsam** [comment]
@robmonte I'm going to close this issue since I cannot reproduce nor have I heard yet from anyone else with this problem. However, if it persists, reply to this issue and I will re-open it.

**Comment by robmcguinness** [comment]
landed on this issue because same is happening to me. `v11.13` works but newer versions do not. I'll see if I can figure out the problem.

**Comment by sfsam** [comment]
https://superuser.com/questions/898124/the-application-someapp-app-can-t-be-opened

I cannot reproduce this so unfortunately I can't test any of the suggestions in the link above. I'm wondering if anything there is helpful.

**Comment by robmcguinness** [comment]
@sfsam the linked provided a solution that worked:

```
chmod -R +x /Applications/Itsycal.app/Contents/MacOS/
```

IMO this can be closed.

**Comment by sfsam** [comment]
I wonder why the executable flag wasn't set. May I ask what you are using to unzip the downloaded .zip file? I'm just grasping at straws trying to understand why chmod fixes it. Do you have any idea?

**Comment by robmcguinness** [comment]
my guess is corporate settings on the laptops :)

**Comment by sfsam** [comment]
Interesting, though I still don't understand why older versions work, but new ones don't. I guess I'll close this again, but will reopen if necessary.

**Comment by robmonte** [comment]
This issue still happens randomly to me. Updating via homebrew never works for me.

As a test, I took the file homebrew downloads - currently it is this URL: https://itsycal.s3.amazonaws.com/Itsycal-0.11.17.zip
vs
the download from your website - currently this URL: https://itsycal.s3.amazonaws.com/Itsycal.zip

and compared their SHA-256 checksums.

Homebrew: a692a283c060c31da7f17ff995cbf9e959e6f74e4c5d0f80484ed666c0e01713
Website: 35378ce992cf246ebb1efc25673ad03f0f8180fd5e650eaa84675bed2d574829

They are different files. On top of that, Homebrew is expecting yet another, third hash:
fda1ba5611deaf4d5b834118b3af37ea9c5d08d1f8c813d04e7dd0552a270e11

No idea what's going on here. On top of that, downloading either of those files manually and trying to run the app file gives me an error. Giving execute permissions to the MacOS folder inside the app bundle does not fix it for me either.

It is also 100% not a corporate setting issue for me.

**Comment by sfsam** [comment]
I don't know anything about Homebrew so I cannot speak to that.

I just downloaded both Itsycal.zip and Itsycal-0.11.17.zip from my website. 
https://itsycal.s3.amazonaws.com/Itsycal.zip
https://itsycal.s3.amazonaws.com/Itsycal-0.11.17.zip

Here is a screenshot of the SHA-256 checksums:
<img width="699" alt="Screen Shot 2019-09-06 at 5 07 21 PM" src="https://user-images.githubusercontent.com/118104/64460630-dd849100-d0c8-11e9-8891-17760308ce1f.png">
They are the same and they match the "third hash" referenced in your message above. 

In the repo you can see the script I use to make these zips. It's just a cp command on line [45](https://github.com/sfsam/Itsycal/blob/40641c94998172e0ca4a6eb2edce521d32e66f04/make_zips_and_appcast.sh#L45) so they really should be the same file. Also in the repo is a doc called [BUILD.md](https://github.com/sfsam/Itsycal/blob/master/BUILD.md) which describes the exact steps I use to build the app. It's just the standard Xcode Archive and Notarization workflow - nothing custom by me.

I don't know why your SHA-256 checksums are different from mine. This is a mystery to me.

**Comment by robmonte** [comment]
Do you use Firefox or Chrome? I don't know why, but that idea popped into my head at seeing your checksums being correct.

So, I downloaded the files from Firefox and sure enough both checksums I calculate are the one you got.

I deleted the files and downloaded them again from Chrome, and it is the Chrome downloads that checksum to the wrong values. This also begs the question, why is the way Homebrew is downloading this (presumably uses Curl, but I really don't know) causing this.

I can't understand why the files would differ at all depending on web browser. It is definitely changing the file contents too, as the ones downloaded from FIrefox actually run and install correctly.

Either way, it's clear now that the issue is outside of Itsycal. However, seeing as I'm no longer the only one to have encountered this issue, it may be happening to others too. It's just very odd that I don't have this problem with literally anything else I use.

**Comment by sfsam** [comment]
In my previous message I had downloaded the files with Safari and also Brave (which is based on Chromium). Both gave the same result. I've now downloaded both files with Google Chrome and I still get the same checksum.

**Comment by gregorymartin** [comment]
This issue seems to persist in Catalina (I'm on 10.15.4), I've tried all the suggestions above with no luck.  Someone did mention "corporate settings" above, which may be related, my company does run JAMF on our machines.

<img width="532" alt="Screen Shot 2020-05-02 at 1 32 14 PM" src="https://user-images.githubusercontent.com/1591110/80891740-f99df980-8c7a-11ea-98d4-96dae7efdd94.png">
